packages/react/src/components/Modal/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { PropsWithChildren, useEffect } from "react";
+import classNames from "classnames";
+import { createPortal } from "react-dom";
+import { Button } from ":/components/Button";
+
+export type ModalHandle = {};
+
+export enum ModalSize {
+  SMALL = "small",
+  MEDIUM = "medium",
+  LARGE = "large",
+  FULL = "full",
+}
+
+export const useModal = ({
+  isOpenDefault,
+}: { isOpenDefault?: boolean } = {}) => {
+  const [isOpen, setIsOpen] = React.useState(!!isOpenDefault);
+  const onClose = () => {
+    setIsOpen(false);
+  };
+
+  const open = () => {
+    setIsOpen(true);
+  };
+
+  const close = () => {
+    onClose();
+  };
+
+  return {
+    isOpen,
+    onClose,
+    open,
+    close,
+  };
+};
+
+export interface ModalProps extends PropsWithChildren {
+  size: ModalSize;
+  isOpen: boolean;
+  onClose: () => void;
+  leftActions?: React.ReactNode;
+  rightActions?: React.ReactNode;
+  actions?: React.ReactNode;
+  title?: string;
+  titleIcon?: React.ReactNode;
+  hideCloseButton?: boolean;
+}
+
+export const Modal = (props: ModalProps) => {
+  const ref = React.useRef<HTMLDialogElement>(null);
+
+  useEffect(() => {
+    if (props.isOpen) {
+      ref.current?.showModal();
+    } else {
+      ref.current?.close();
+    }
+  }, [props.isOpen]);
+
+  useEffect(() => {
+    ref.current?.addEventListener("close", () => {
+      props.onClose();
+    });
+  }, []);
```

Comment:
You must remove the event listener on unmount.

```suggestion
  useEffect(() => {
    ref.current?.addEventListener("close", props.onClose);
    
    return () => {
	    ref.current?.removeEventListener("close", props.onClose);
    }
  }, []);
```

or as this event is triggered on modal close, you could maybe take apart of `once` options.

```suggestion
  useEffect(() => {
    ref.current?.addEventListener("close", props.onClose, { once: true });
  }, []);
```